numba/core/types/common.py

```diff
@@ -49,7 +49,10 @@ def __init__(self, dtype, ndim, layout, readonly=False, name=None):
         from .misc import unliteral
 
         if isinstance(dtype, Buffer):
-            raise TypeError("Buffer dtype cannot be buffer, have dtype: {}".format(dtype))
+            msg = ("The dtype of a Buffer type cannot itself be a Buffer type, "
+                   "this is unsupported behaviour."
+                   "\nThe dtype requested for the offending Buffer was: {}.")
```

Comment:
maybe switch "offending" to "unsupported"?  Seems less harsh.  :)

Reply:
thanks, done in 
d68d688